### PR TITLE
Adding status and lastPoll fields to CrStates

### LINKED ISSUE #6448: Add the cache status field to the Traffic Monitor CrStates endpoint
## This Feature Request affects these Traffic Control components:

- Traffic Monitor

## Description
Currently the tc-health-client is using the Traffic Monitor api/cache-statuses endpoint to find and mark down unhealthy parents.  This endpoint it turns out is for local TM state and may not reflect the state as reached by consensus of all TM's.  The health-client is being switched to use the publish/CrStates endpoint instead.

However, the publish/CrStates endpoint lacks the **status** field that is available in the api/cache-statuses endpoint.  This field was logged by the health-client so that it may be understood why the parent was marked down ie, the health threshold that triggered the markdown.  It would be invaluable to continue this logging if the field could be added to the publish/CrStates endpoint or a query parameter that would let the client ask for the field as part of the response.  Querying both endpoints is not desirable.

Also, a date time field should be added to the publish/Crstates endpoint response that shows when the health monitor last polled the cache.  This could also be added to the response only when requested with a query parameter.

Example **status** information from api/cache-statuses:

- **REPORTED - availableBandwidthInKbps too low (0.00 < 75000000.00**
- **REPORTED - available**
- **REPORTED - queryTime too high (1520.00 > 1500.00)**

Possible new struct layout, see lib/go-tc/crstats.go:
```
type IsAvailable struct  {
        IsAvailable bool  
        Ipv4Available bool
        Ipv6Available bool   
        Status string // new field status and health threshold reason when unavailable
        LastHmPoll time.Time // new field last Health monitor poll` 
}
```

### PR DESCRIPTION
**Pull request by srijeet0406**
This PR closes #6448 

<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Monitor

## What is the best way to verify this PR?

Run TM locally and `GET` the `publish/CrStates` endpoint. Make sure the `status` and `lastPoll` fields show up in the output. Also confirm that the status for the caches is in fact what's being reported by the `api/cache-statuses` endpoint.

## If this is a bugfix, which Traffic Control versions contained the bug?

- master

## PR submission checklist
- [ ] This PR has tests 
- [x] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Review by ocket8888 — COMMENTED** [review]
Would you want to take this opportunity to fill out [our missing docs for `/publish/CrStates`](https://traffic-control-cdn.readthedocs.io/en/latest/development/traffic_monitor/traffic_monitor_api.html#publish-crstates)?

File: lib/go-tc/crstates.go [hunk 1]
```diff
@@ -37,10 +38,12 @@ type CRStatesDeliveryService struct {
...
-	IsAvailable    bool `json:"isAvailable"`
-	Ipv4Available  bool `json:"ipv4Available"`
-	Ipv6Available  bool `json:"ipv6Available"`
-	DirectlyPolled bool `json:"-"`
+	IsAvailable    bool      `json:"isAvailable"`
+	Ipv4Available  bool      `json:"ipv4Available"`
+	Ipv6Available  bool      `json:"ipv6Available"`
+	DirectlyPolled bool      `json:"-"`
+	Status         string    `json:"status"`
+	LastHmPoll     time.Time `json:"lastHmPoll"`
```
Comment (rawlinp) [inline lib/go-tc/crstates.go:46]: Can we just call this `LastPoll`? The `Hm` bit is kind of unnecessary/already understood.

File: docs/source/development/traffic_monitor/traffic_monitor_api.rst [hunk 2]
```diff
@@ -343,22 +343,60 @@ TODO
...
 
 Response Structure
 """"""""""""""""""
+:caches: An object with keys that are the names of monitored :term:`cache servers`
+	:server name: The name of the server being monitored
 
-TODO
+		:isAvailable: Whether or not this cache is available for monitoring
+		:ipv4Available: Whether or not an IPV4 interface on this cache is available for monitoring
+		:ipv6Available: Whether or not an IPV6 interface on this cache is available for monitoring
```
Comment (rawlinp) [inline docs/source/development/traffic_monitor/traffic_monitor_api.rst:366]: "available for ~monitoring~ routing" would make more sense I think

File: docs/source/development/traffic_monitor/traffic_monitor_api.rst [hunk 3]
```diff
@@ -343,22 +343,60 @@ TODO
...
+		:status: The status of this server, along with any additional reason for it to be marked as such
+		:lastHmPoll: The last time the health data for this server was polled by a traffic monitor
 
+:deliveryServices: An object with keys as the names of monitored :term:`Delivery Services`
+	:delivery service name: The name of the delivery service being monitored
+		:disabledLocations: A list of disabled locations for this delivery service
+		:isAvailable: Whether or not this delivery service is available for monitoring
```
Comment (rawlinp) [inline docs/source/development/traffic_monitor/traffic_monitor_api.rst:373]: "available for ~monitoring~ routing" would make more sense I think

**Review by rawlinp — CHANGES_REQUESTED** [review]

File: docs/source/development/traffic_monitor/traffic_monitor_api.rst [hunk 4]
```diff
@@ -343,22 +343,60 @@ TODO
...
+	:caption: Example Request
+
+	GET /publish/CrStates HTTP/1.1
+	Accept: */*
+	Content-Type: application/json
 
 Response Structure
 """"""""""""""""""
+:caches: An object with keys that are the names of monitored :term:`cache servers`
+	:server name: The name of the server being monitored
```
Comment (rawlinp) [inline docs/source/development/traffic_monitor/traffic_monitor_api.rst:362]: I think there needs to be an empty line between indentation levels for this to render properly

File: docs/source/development/traffic_monitor/traffic_monitor_api.rst [hunk 5]
```diff
@@ -343,22 +343,60 @@ TODO
...
 
-TODO
+		:isAvailable: Whether or not this cache is available for routing
+		:ipv4Available: Whether or not an IPV4 interface on this cache is available for routing
+		:ipv6Available: Whether or not an IPV6 interface on this cache is available for routing
+		:status: The status of this server, along with any additional reason for it to be marked as such
+		:lastPoll: The last time the health data for this server was polled by a traffic monitor
 
+:deliveryServices: An object with keys as the names of monitored :term:`Delivery Services`
+	:delivery service name: The name of the delivery service being monitored
```
Comment (rawlinp) [inline docs/source/development/traffic_monitor/traffic_monitor_api.rst:371]: I think there needs to be an empty line between indentation levels for this to render properly

**Review by rawlinp — CHANGES_REQUESTED** [review]

**Review by rawlinp — APPROVED** [review]

**Pull request merged**